File: leetcode-go/169.Majority-Element/majority_element.go
package main

import "sort"

/*
 * @lc app=leetcode.cn id=169 lang=golang
 *
 * [169] 多数元素
 */

// @lc code=start
func majorityElement(nums []int) int {
	m := make(map[int]int)
	for _, v := range nums {
		m[v]++
	}
	for k, v := range m {
		if v > len(nums)/2 {
			return k
		}
	}
	return -1
}

// @lc code=end

func majorityElement1(nums []int) int {
	sort.Ints(nums)
	return nums[len(nums)/2]
}

File: leetcode-go/169.Majority-Element/majority_element.js
/*
 * @lc app=leetcode.cn id=169 lang=javascript
 *
 * [169] 多数元素
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  nums.sort();
  return nums[Math.floor(nums.length / 2)];
};
// @lc code=end

var majorityElement1 = function (nums) {
  let map = new Map();
  for (const value of nums) {
    if (map.has(value)) {
      map.set(value, map.get(value) + 1);
    } else {
      map.set(value, 1);
    }
  }

  for (const [k, v] of map.entries()) {
    if (v > nums.length / 2) {
      return k;
    }
  }
  return -1;
};
